cas-compute/src/symbolic/derivative.rs

```diff
@@ -0,0 +1,268 @@
+use rug::Integer;
+
+use crate::primitive::{float, int};
+
+use super::expr::Primary;
+use super::Expr;
+
+// Something that is not trivially zero - intended to clean up ASTs and not mathetmatically
+// rigorous
+fn non_zero(e: &Expr) -> bool {
```

Comment:
```suggestion
/// Returns `true` if the given [`Expr`] is "clearly" nonzero. This is intended to clean up ASTs
/// and is not mathetmatically rigorous.
fn is_non_zero(e: &Expr) -> bool {
```

I made this a doc-comment and rewrote the docs to be closer to my preferred style.